R/merge.R

```diff
@@ -52,32 +52,41 @@
 
 #' @importFrom S4Vectors SimpleList
 #' @importFrom scuttle sumCountsAcrossFeatures
-.merge_rows <- function(x, f, archetype = 1L, 
+.merge_rows <- function(x, f, archetype = 1L,
+                        na.rm = FALSE,
                         average = FALSE,
                         BPPARAM = SerialParam(),
                         check.assays = TRUE,
                         ...){
     # input check
+    if(!.is_a_bool(na.rm)){
+        stop("'na.rm' must be TRUE or FALSE.", call. = FALSE)
+    }
     if( !.is_a_bool(average) ){
         stop("'average' must be TRUE or FALSE.", call. = FALSE)
     }
     if( !.is_a_bool(check.assays) ){
         stop("'check.assays' must be TRUE or FALSE.", call. = FALSE)
     }
-    #
+
+    # This is done otherwise we lose NA values
+    if ( na.rm ) {
+        f[is.na(f)] <- "NA"
+    }
     if(is.character(f) && length(f)==1 && f %in% colnames(rowData(x))){
         f <- factor(as.character(rowData(x)[, f]))
     }
     else if(is.character(f) && length(f)==1 && f %in% colnames(colData(x))){
+        f[is.na(f)] <- "NA"
```

Comment:
This does na.rm without asking from user

Reply:
Also, I am wondering what was the intention of having colData here. This is merge_rows.... We should probably fix this